test/interval_test.exs

```diff
@@ -30,6 +88,11 @@ defmodule IntervalTests do
     assert ["15:00", "15:10", "15:20", "15:30", "15:40", "15:50"] == steps
   end
 
+  test "halts when enumerating with an invalid step unit" do
+    interval = %Interval{from: ~D[2017-04-02], until: ~D[2017-05-02], step: [invalid_step: 1]}
+    assert 0 == Enum.count(interval)
```

Comment:
I'd rather get an error raised here. From my point of view, returning 0 in this situation may be misleading and possibly hard to debug. "Fail fast", that sort of things. What do you think?